quic/api/test/QuicTransportBaseTest.cpp

```diff
@@ -3678,5 +3678,22 @@ TEST_F(QuicTransportImplTest, ImplementationObserverCallbacksInvoked) {
   EXPECT_EQ(0, size(transport->transportConn->pendingCallbacks));
 }
 
+TEST_F(
+    QuicTransportImplTest,
+    ImplementationObserverCallbacksCorrectQuicSocket) {
+  void* returnedSocket = nullptr;
+  auto func = [&](QuicSocket* qSocket) { returnedSocket = qSocket; };
```

Comment:
You already seem to be using `QuicSocket*` here -- what's the reason for `void*`?

Reply:
Same as above.